### PR TITLE
feat: v8::Context::set_promise_hooks accepts None

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

**Comment by aapoalas** [comment]
LGTM

**Review by littledivy — APPROVED** [review]
LGTM

**Review by piscisaureus — APPROVED** [review]
You have to update `v8__Context__SetPromiseHooks` in binding.cc.
They cannot take a value by reference as they currently do.

**Review by piscisaureus — CHANGES_REQUESTED** [review]
You have to update `v8__Context__SetPromiseHooks` in binding.cc.
Its parameters are currently of a reference type `v8::Function&`. You can't assign nullptr to those.

File: src/context.rs [hunk 1]
```diff
@@ -110,18 +110,18 @@ impl Context {
...
       v8__Context__SetPromiseHooks(
         self,
-        &*init_hook,
-        &*before_hook,
-        &*after_hook,
-        &*resolve_hook,
+        init_hook.map_or_else(std::ptr::null, |v| &*v),
+        before_hook.map_or_else(std::ptr::null, |v| &*v),
+        after_hook.map_or_else(std::ptr::null, |v| &*v),
+        resolve_hook.map_or_else(std::ptr::null, |v| &*v),
```
Comment (piscisaureus) [inline src/context.rs:124]:
```suggestion
        init_hook.map_or_else(null, |v| &*v),
        before_hook.map_or_else(null, |v| &*v),
        after_hook.map_or_else(null, |v| &*v),
        resolve_hook.map_or_else(null, |v| &*v),
```

File: src/context.rs [hunk 2]
```diff
@@ -110,18 +110,18 @@ impl Context {
   #[inline(always)]
   pub fn set_promise_hooks(
     &self,
```
Comment (piscisaureus) [inline src/context.rs:112]: A problem not introduced by your PR, however for posterity:
An API that mutates the `Isolate` or any object on the V8 heap should have at least one `&mut` parameter. 
Usually that would be an `&mut HandleScope` but if that's not possible somehow there must be something else.

File: src/binding.cc [hunk 3]
```diff
@@ -1709,13 +1709,13 @@ const v8::Object* v8__Context__GetExtrasBindingObject(v8::Context& self) {
...
 }
 
-void v8__Context__SetPromiseHooks(v8::Context& self, v8::Function& init_hook,
-                                  v8::Function& before_hook,
-                                  v8::Function& after_hook,
-                                  v8::Function& resolve_hook) {
+void v8__Context__SetPromiseHooks(v8::Context& self, v8::Function* init_hook,
+                                  v8::Function* before_hook,
+                                  v8::Function* after_hook,
+                                  v8::Function* resolve_hook) {
```
Comment (piscisaureus) [inline src/binding.cc:1715]:
```suggestion
void v8__Context__SetPromiseHooks(v8::Context& self,
                                  const v8::Function* init_hook,
                                  const v8::Function* before_hook,
                                  const v8::Function* after_hook,
                                  const v8::Function* resolve_hook) {
```

**Pull request merged**